aikau/src/main/resources/alfresco/layout/VerticalReveal.js

```diff
@@ -83,61 +92,94 @@ define(["alfresco/core/ProcessWidgets",
       initiallyRevealed: false,
 
       /**
+       * Will be resolved once the widgets have been processed
        *
-       * 
        * @instance
+       * @type {object}
        */
-      postMixInProperties: function alfresco_layout_VerticalReveal__postMixInProperties() {
-         if (this.toggleLabel == null)
-         {
-            // In the postCreate we'll hide the toggle node...
+      _widgetProcessingDeferred: null,
+
+      /**
+       * Called when the child widgets have been processed
+       *
+       * @instance
+       * @override
+       * @param {Array} widgets An array of all the widgets that have been processed
+       */
+      allWidgetsProcessed: function alfresco_layout_VerticalReveal__allWidgetsProcessed( /*jshint unused:false*/ widgets) {
+         this._widgetProcessingDeferred.resolve();
+      },
+
+      /**
+       * Widget has been created, but not sub-widgets
+       *
+       * @instance
+       * @override
+       */
+      postCreate: function alfresco_layout_VerticalReveal__postCreate() {
+         if (this.toggleLabel) {
+            domClass.add(this.domNode, this.rootClass + "--has-toggle");
          }
-         else
-         {
-            this.toggleLabel = this.message(this.toggleLabel);
+         this.alfSubscribe(this.subscriptionTopic, lang.hitch(this, this._onDisplayToggle));
+         if (this.initiallyRevealed === true) {
+            this._processWidgets();
          }
       },
 
       /**
+       * Properties have been mixed into this widget instance.
        *
-       * 
        * @instance
+       * @override
        */
-      postCreate: function alfresco_layout_VerticalReveal__postCreates() {
-         // Set a boolean flag to record when the child widgets get processed so that we only process them once...
-         this._processedWidgetsToReveal = false;
-         
-         if (this.toggleLabel == null)
-         {
-            domClass.add(this.toggleNode, "hidden");
+      postMixInProperties: function alfresco_layout_VerticalReveal__postMixInProperties() {
+         if (this.toggleLabel) {
+            this.toggleLabel = this.message(this.toggleLabel);
          }
-         this.alfSubscribe(this.subscriptionTopic, lang.hitch(this, this.onDisplayToggle));
+         this.inherited(arguments);
+      },
 
-         if (this.initiallyRevealed === true && this.widgets)
-         {
-            this.processWidgets(this.widgets, this.containerNode);
-            this._processedWidgetsToReveal = true;
-         }
+      /**
+       * Handle requests to toggle the contents visibility
+       *
+       * @instance
+       */
+      _onDisplayToggle: function alfresco_layout_VerticalReveal___onDisplayToggle() {
+         this._processWidgets().then(lang.hitch(this, function() {
```

Comment:
We should avoid the use of the closure here, this is likely to be a prime example of something that might want to be overridden by an extending module.